MekHQ/src/mekhq/campaign/personnel/Person.java

```diff
@@ -831,6 +832,104 @@ public boolean hasRole(int role) {
         return (getPrimaryRole() == role) || (getSecondaryRole() == role);
     }
 
+    /**
+     * This is used to determine whether the person has either a primary or secondary role within the
+     * two bounds. This is inclusive of the two bounds
+     * @param minimumRole the minimum role bound (inclusive)
+     * @param maximumRole the maximum role bound (inclusive)
+     * @return true if they have a role within the bounds (inclusive), otherwise false.
+     */
+    public boolean hasRoleWithin(int minimumRole, int maximumRole) {
```

Comment:
I'd almost suggest abstracting these, but it looks like you've got further plans for improvement in this area, so we can leave it as is for now.

Reply:
I do, this is just a step towards the final plans (likely using an enum, but abstraction is under consideration)